streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -71,7 +75,65 @@ public StateStore getStateStore(final String name) {
 
         final StateStore global = stateManager.getGlobalStore(name);
         if (global != null) {
-            return global;
+            return new KeyValueStore() {
```

Comment:
From every usage of `getStateStore` I saw in the source code, the result is cast to `KeyValueStore`. 
And `ReadOnlyKeyValueStore` don't implement `KeyValueStore`.

`WordCountProcessorDemo.java` line 80: 
```
this.kvStore = (KeyValueStore<String, Integer>) context.getStateStore("Counts");
```

I think many users used the same technique to program their Processors. 
So users will obtain the runtime exception in the code that was correct in the previous version.

Reply:
Yes. That is what I meant by

> frankly speaking I am not sure, because it would not be backward compatible... 

It would be cleaner if we could return `ReadOnlyKeyValueStore`, but as a matter of fact, I also believe the most people cast to `KeyValueStore` -- not sure if we could argue that they wrote incorrect code. It was never documented...